puppet/lib/puppet/indirector/facts/puppetdb.rb

```diff
@@ -26,7 +26,15 @@ def save(request)
         facts = request.instance.dup
         facts.values = facts.values.dup
         facts.values[:trusted] = get_trusted_info(request.node)
-        {
+
+        inventory = facts.values['_puppet_inventory_1']
+        package_inventory = nil
+        if inventory && inventory.respond_to?(:keys)
+          package_inventory = inventory['packages']
+          facts.values.delete('_puppet_inventory_1')
```

Comment:
we should always be deleting this if it exists -- respond_to? keys is irrelevant